logical/response.go

```diff
@@ -23,6 +23,11 @@ const (
 	// This can only be specified for non-secrets, and should should be similarly
 	// avoided like the HTTPContentType. The value must be an integer.
 	HTTPStatusCode = "http_status_code"
+
+	// HTTPHeaders is a mechanism to support returning HTTP headers with the response.
+	// This can only be specified for non-secrets, and should should be similarly
+	// avoided like the HTTPContentType. The value must be a map of string to string.
+	HTTPHeaders = "http_headers"
```

Comment:
This isn't the right mechanism for this. If we want to allow arbitrary headers to be returned in a response, it should be added as an object to `logical.Response`, with appropriate rules around HMACing in auditing and so on.